### PR TITLE
Feat/rtk requests for admin user profile

### PR DESCRIPTION
**Pull request by Arseniia-Damaksina**
## Describe your changes

- I added rtk requests to get user, update user, softly delete by id;
- Fixed styling issues on the admin panel;
- Added functionality to all buttons;
- Dynamically render titles and subtitles

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-67)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
demo1
demo2

### CODE REVIEW

File: src/pages/admin/ModalPopup/index.tsx [hunk 1]
```diff
@@ -21,10 +30,41 @@ import {
...
 
-function ModalPopup({ onClose }: IModalPopup) {
+function ModalPopup({ onClose, userId }: IModalPopup) {
   const { t } = useTranslation();
+  const navigate = useNavigate();
+  const { showToast } = useToast();
+
+  const [deleteUserByAdmin] = useDeleteUserByAdminMutation();
+
+  async function handleDeleteUser(id: string | undefined) {
```
Comment (lyudad) [inline src/pages/admin/ModalPopup/index.tsx]: handleDeleteUser(id?: string)  should work

File: src/pages/admin/UsersPage/UsersTable/index.tsx [hunk 2]
```diff
@@ -12,6 +14,8 @@ import formatDateString from 'src/common/formatDateString';
 import { IAdminUser } from 'src/redux/user/types';
 
 import ActionButtons from '../../ActionButtons';
+import ModalPopup from '../../ModalPopup';
+import StyledBackdrop from '../../StyledBackdrop';
 import StyledPagination from '../../StyledPagination';
```
Comment (lyudad) [inline src/pages/admin/UsersPage/UsersTable/index.tsx]: alias

File: src/pages/admin/UsersPage/index.tsx [hunk 3]
```diff
@@ -102,12 +95,21 @@ function UsersPage() {
     return <CircularProgress />;
   }
 
+  const getTitle = (path: string) => {
```
Comment (lyudad) [inline src/pages/admin/UsersPage/index.tsx]: return param

File: src/common/extractPhoneNumberParts.ts
Comment (lyudad) [inline src/common/extractPhoneNumberParts.ts:1]: create one more folder under /common, call it "utils" => moe all reusable functions there

File: src/pages/admin/ModalPopup/index.tsx (hunk 1 above)
Comment (Arseniia-Damaksina) [reply src/pages/admin/ModalPopup/index.tsx]: fixed

File: src/pages/admin/UsersPage/UsersTable/index.tsx (hunk 2 above)
Comment (Arseniia-Damaksina) [reply src/pages/admin/UsersPage/UsersTable/index.tsx]: fixed

File: src/pages/admin/UsersPage/index.tsx (hunk 3 above)
Comment (Arseniia-Damaksina) [reply src/pages/admin/UsersPage/index.tsx]: fixed

File: src/common/extractPhoneNumberParts.ts
Comment (Arseniia-Damaksina) [reply src/common/extractPhoneNumberParts.ts:1]: done

**Pull request merged**